### PR TITLE
Providerize formation

### PR DESCRIPTION
**Pull request by ddollar**
Moves Formation into the provider interface

### CODE REVIEW

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;ZvLYaspi&#x2F;558-provider-formation"> Provider: Formation</a>

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;XdljMupW&#x2F;577-formation-provider"> formation provider</a>

File: provider/aws/apps.go [hunk 1]
```diff
@@ -19,37 +18,24 @@ func (p *AWSProvider) AppGet(name string) (*structs.App, error) {
...
-		})
-	} else {
-
-		// try '$RACK-myapp', and if not found try 'myapp'
-		res, err = p.describeStacks(&cloudformation.DescribeStacksInput{
-			StackName: aws.String(os.Getenv("RACK") + "-" + name),
-		})
-
-		if awsError(err) == "ValidationError" {
-			res, err = p.describeStacks(&cloudformation.DescribeStacksInput{
```
Comment (nzoschke) [inline provider/aws/apps.go]: are you intentionally no longer looking up old naming format app stacks?

File: vendor/github.com/convox/nlogger/nlogger.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package nlogger
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/ddollar/logger"
```
Comment (nzoschke) [inline vendor/github.com/convox/nlogger/nlogger.go]: can this use convox/logger?

File: provider/aws/apps.go (hunk 1 above)
Comment (ddollar) [inline provider/aws/apps.go]: Yes. An older release completely dropped support for deploying them. Anyone that really needs to stay on that format can use an older Rack release.

**Pull request closed without merging**